### PR TITLE
Add winforms desktop application

### PR DESCRIPTION
**Pull request by olivercoad**
Adds a winforms project with installer to register as windows explorer shell extension.

### CODE REVIEW

**Comment by cdilga** [comment]
Tested MSI which correctly functions for each of the hash types, though it's slower to retrieve images than I'd expect. Inference time is 0.06s, measured on the generator network, though a request takes approximately 25-50s to be returned, despite https://checkface.ml working in much less time

File: src/dotnet-windows/checkface-dotnet/CheckFaceForm.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            if (String.IsNullOrWhiteSpace(textBox1.Text))
+            {
+                pictureBox1.Image = Properties.Resources.face_transparent;
+            }
+            else
+            {
+                pictureBox1.Image = Properties.Resources.loader;
+                pictureBox1.SizeMode = PictureBoxSizeMode.CenterImage;
+                string enc = System.Web.HttpUtility.UrlEncode(textBox1.Text);
+                pictureBox1.LoadAsync($"https://api.checkface.ml/api/{enc}");
```
Comment (cdilga) [inline src/dotnet-windows/checkface-dotnet/CheckFaceForm.cs]: Perhaps adding `?dim={dimension}` to the end would resolve the extreme slowness

Comment (olivercoad) [reply src/dotnet-windows/checkface-dotnet/CheckFaceForm.cs]: It defaults to 300, which is why it's blurry if you drag it larger. But it's probably a good idea to specify it anyway.

**Comment by olivercoad** [comment]
> Tested MSI which correctly functions for each of the hash types, though it's slower to retrieve images than I'd expect. Inference time is 0.06s, measured on the generator network, though a request takes approximately 25-50s to be returned, despite https://checkface.ml working in much less time

Will try handling the web requests manually rather than relying on the `PictureBox.LoadAsync()` to do it.

**Pull request merged**